src/services/azureFunctionsService.ts

```diff
@@ -104,6 +104,10 @@ export class AzureFunctionsService implements mssql.IAzureFunctionsService {
 			folderPath: projectFile
 		});
 
+		// check for the new function file to be created and dispose of the file system watcher
+		const timeout = timeoutPromise(LocalizedConstants.timeoutAzureFunctionFileError);
+		const functionFile = await Promise.race([newFunctionFileObject.filePromise, timeout]).finally(newFunctionFileObject.watcherDisposable.dispose());
```

Comment:
Actually - let's wrap lines 87 - 109 in a try/finally to ensure that we clean the watcher up even if something else happens like the user exits the flow.